### ISSUE TITLE
Interface 'FetchResult<TData, C, E>' incorrectly extends interface 'ExecutionResult<ObjMap<unknown>, ObjMap<unknown>>

### ISSUE THREAD

**Issue by gajus** [apollographql/apollo-client#9034]
**Intended outcome:**

No TypeScript errors.

**Actual outcome:**

TypeScript error.

```
node_modules/@apollo/client/link/core/types.d.ts:20:18 - error TS2430: Interface 'FetchResult<TData, C, E>' incorrectly extends interface 'ExecutionResult<ObjMap<unknown>, ObjMap<unknown>>'.
  Types of property 'data' are incompatible.
    Type 'TData | null | undefined' is not assignable to type 'ObjMap<unknown> | null | undefined'.
      Type 'TData' is not assignable to type 'ObjMap<unknown> | null | undefined'.
        Type 'TData' is not assignable to type 'ObjMap<unknown>'.

20 export interface FetchResult<TData = {
```

**How to reproduce the issue:**

Use `tsc` with `{"skipLibCheck": false}`

**Versions**

```
  System:
    OS: macOS 11.6
  Binaries:
    Node: 16.11.1 - ~/.nvm/versions/node/v16.11.1/bin/node
    Yarn: 1.22.17 - ~/.nvm/versions/node/v16.11.1/bin/yarn
    npm: 8.0.0 - ~/.nvm/versions/node/v16.11.1/bin/npm
  Browsers:
    Chrome: 95.0.4638.69
    Firefox: 92.0
    Safari: 14.1.2
  npmPackages:
    @apollo/client: ^3.4.17 => 3.4.17
    @apollo/link-error: ^2.0.0-beta.3 => 2.0.0-beta.3
    @apollo/react-hooks: ^4.0.0 => 4.0.0
    @apollo/react-ssr: ^4.0.0 => 4.0.0
    apollo-link-sentry: ^3.0.2 => 3.0.2
```

**Comment by brainkim** [comment]
@gajus What’s your graphql-js version? These are types from graphql-js.

**Comment by juane1000** [comment]
I am also experiencing this after the most recent update. These updates were both apollo-client and graphql (to 16.0.1).

**Comment by gajus** [comment]
Same here.

**Comment by brainkim** [comment]
@juane1000 @gajus The package version numbers I need for this specific issue are:

`@apollo/client`, `graphql`, `typescript`.

**Comment by juane1000** [comment]
> @juane1000 @gajus The package version numbers I need for this specific issue are:
> 
> `@apollo/client`, `graphql`, `typescript`.

I reverted back to graphql@15.7.2 and that resolved the issue using the latest apollo-client version.

To answer your question: @apollo/client@3.4.17 , graphql@16.0.1 , typescript@4.4.4

**Comment by juane1000** [comment]
@brainkim I do see that @apollo/client has the peer-dep between 14 & 15 for graphql. It was my mistake for using the ignore-legacy-deps flag for the npm updates I was doing.

**Comment by brainkim** [comment]
Right, so Apollo Client ^3.5 is the first version with support for GraphQL 16. Please feel to try that out if you wish.

**Comment by brainkim** [comment]
I also just think `skipLibCheck` should always be true but that’s just me.

**Comment by annienero** [comment]
Seeing another typescript issue with ExecutionResult- "Generic type 'ExecutionResult<TData>' requires between 0 and 1 type arguments." Let me know if I should open a separate issue.
<img width="1392" alt="Screen Shot 2021-12-07 at 6 54 30 PM" src="https://user-images.githubusercontent.com/22509306/145123908-8ea51041-318f-4bc1-867f-c1ea2b6e1684.png">

with apollo/client 3.5.5, graphql: 14.5.8, and typescript 4.0.0

**Comment by pawelidziak** [comment]
Hey, I have tried to install apollo with angular 13, and for the newest packages versions, I face the same error.  I tried both ways for install (`ng add apollo-angular` and `npm i` packages separately, how is presented in [Get started](https://apollo-angular.com/docs/get-started))

![image](https://user-images.githubusercontent.com/22195238/151661182-56fd8547-8b9d-4f14-838b-8e09b6400159.png)

Packages version:

``` 
"@angular/cli": "~13.2.0",
...
"@apollo/client": "^3.5.8",
"apollo-angular": "^2.6.0",
"graphql": "^16.3.0",
"typescript": "~4.5.2"
```

I have tried to downgrade apollo-client or graphql package but nothing works. Have you guys some advice on how to solve it?

**Comment by lano-vargas** [comment]
Had same issue. After downgrading "graphql": "^16.3.0", TO "graphql": "^15.0.0",  seem to have worked
